drivers/gles3/rasterizer_scene_gles3.cpp

```diff
@@ -5168,7 +5173,7 @@ void RasterizerSceneGLES3::initialize() {
 		glBufferData(GL_UNIFORM_BUFFER, sizeof(LightDataUBO), NULL, GL_DYNAMIC_DRAW);
 		glBindBuffer(GL_UNIFORM_BUFFER, 0);
 
-		state.max_forward_lights_per_object = 8;
+		state.max_forward_lights_per_object = render_list.max_lights_per_object;
```

Comment:
This should be ``MIN(state.max_ubo_lights, render_list.max_lights_per_object)``

The majority of devices only support a ``GL_MAX_UNIFORM_BLOCK_SIZE`` of 65536. This is true of both mobile and desktop. With a ``GL_MAX_UNIFORM_BLOCK_SIZE`` of 65536 we can have at most 65536/160 = 409 lights in the scene. A large proportion of devices only support a ``GL_MAX_UNIFORM_BLOCK_SIZE`` of 16384, which is 102 lights per scene. Since the ``max_lights_per_object`` can be so much higher, it is a good idea to limit it.

Reply:
ok ill go ahead and make the changes

Reply:
done